src/traces/bar/layout_defaults.js

```diff
@@ -46,7 +48,6 @@ module.exports = function(layoutIn, layoutOut, fullData) {
 
     if(!hasBars) return;
```

Comment:
we should 

```js
delete layoutOut.barmode;
```

when `hasBars` is false, so that it doesn't pollute `fullLayout`.